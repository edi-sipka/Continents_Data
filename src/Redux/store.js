import {
  configureStore,
  applyMiddleware,
  combineReducers,
} from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import countryReducers from './reducers';
import categorySelector from './category';

const rootReducer = combineReducers({
  countries: countryReducers,
  category: categorySelector,
});

const store = configureStore(
  { reducer: rootReducer },
  composeWithDevTools(applyMiddleware(thunk)),
);
export default store;
