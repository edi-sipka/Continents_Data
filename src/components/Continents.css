.list {
  list-style: none;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 2rem;
  padding: 3px;
  margin-top: 50px;
}

.btn {
  color: white;
}

.position {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  margin-left: 10px;
}

.button {
  background-color: #af059654;
  color: white;
  padding: 12px;
  width: 330px;
  margin-left: 20px;
  margin-top: 40px;
  border: none;
  font-size: 20px;
  box-shadow: 0 5px 25px rgba(0, 0, 0, 0.2);
  -webkit-appearance: button;
  appearance: button;
  outline: none;
  font-family: 'Lato', sans-serif;
}

.lists {
  background-color: rgba(35, 1, 36, 0.43);
  box-shadow: 5px 10px #aeaeae4e;
  display: flex;
  margin: 20px;
}

.lists:hover {
  transform: scale(1.1);
  transition: all 0.5s ease-in;
}
