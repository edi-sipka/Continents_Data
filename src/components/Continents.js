/* eslint-disable */
import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getList } from '../Redux/reducers';
import continents from '../list/list';
import { useParams } from 'react-router';
import { BsFillArrowRightCircleFill } from 'react-icons/bs';
import { Link } from 'react-router-dom';
import './Continents.css';
function Continents() {
  const dispatch = useDispatch();
  const { idName } = useParams();
  const listOfCountries = useSelector((state) => state.countries);
  const countriesList = (name) => {
    dispatch(getList(name));
  };
  return (
    <div>
      <div className="section">
        {continents.map(({ id, name, image }) => (
          <li key={id}>
            <Link to={`${name}`}>{name}</Link>
            <img src={image} />
            <button onClick={() => countriesList(name)} type="button">
              <BsFillArrowRightCircleFill />
            </button>
          </li>
        ))}
      </div>
    </div>
  );
}

export default Continents;
