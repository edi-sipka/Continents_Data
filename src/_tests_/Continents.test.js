/* eslint-disable */

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Continents from '../components/Continents';

describe('Navbar rendering', () => {
  test('Navbar rendering', () => {
    const render = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Continents />
        </BrowserRouter>
      </Provider>
    );
    expect(render).toMatchSnapshot();
  });
});
