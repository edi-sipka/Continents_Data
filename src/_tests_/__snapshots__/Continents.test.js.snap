// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar rendering Navbar rendering 1`] = `
<div
  className="section"
>
  <label
    htmlFor="category"
  >
    <select
      className="button"
      name="category"
      onChange={[Function]}
      placeholder="Category"
      type="text"
      value="Select Continent"
    >
      <option
        value="Select Continent"
      >
        SELECT YOUR CONTINENT
      </option>
      <option
        value="All"
      >
        SHOW ALL
      </option>
      <option
        value="Africa"
      >
        AFRICA
      </option>
      <option
        value="Asia"
      >
        ASIA
      </option>
      <option
        value="Antarctica"
      >
        ANTARCTICA
      </option>
      <option
        value="Europe"
      >
        EUROPE
      </option>
      <option
        value="Oceania"
      >
        OCEANIA
      </option>
      <option
        value="North America"
      >
        NORTH AMERICA
      </option>
      <option
        value="South America"
      >
        SOUTH AMERICA
      </option>
    </select>
  </label>
  <ul
    className="list"
  >
    <li
      className="lists"
    >
      <img
        alt="countries"
        src="/Images/Europe.svg"
      />
      <div
        className="position"
      >
        <h2
          className="title-name"
        >
          Europe
        </h2>
        <a
          href="/Europe"
          onClick={[Function]}
        >
          <svg
            className="btn"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
            />
          </svg>
        </a>
      </div>
    </li>
    <li
      className="lists"
    >
      <img
        alt="countries"
        src="/Images/Africa.svg"
      />
      <div
        className="position"
      >
        <h2
          className="title-name"
        >
          Africa
        </h2>
        <a
          href="/Africa"
          onClick={[Function]}
        >
          <svg
            className="btn"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
            />
          </svg>
        </a>
      </div>
    </li>
    <li
      className="lists"
    >
      <img
        alt="countries"
        src="/Images/Antartica.svg"
      />
      <div
        className="position"
      >
        <h2
          className="title-name"
        >
          Antartica
        </h2>
        <a
          href="/Antartica"
          onClick={[Function]}
        >
          <svg
            className="btn"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
            />
          </svg>
        </a>
      </div>
    </li>
    <li
      className="lists"
    >
      <img
        alt="countries"
        src="/Images/Asia.svg"
      />
      <div
        className="position"
      >
        <h2
          className="title-name"
        >
          Asia
        </h2>
        <a
          href="/Asia"
          onClick={[Function]}
        >
          <svg
            className="btn"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
            />
          </svg>
        </a>
      </div>
    </li>
    <li
      className="lists"
    >
      <img
        alt="countries"
        src="/Images/North-America.svg"
      />
      <div
        className="position"
      >
        <h2
          className="title-name"
        >
          North America
        </h2>
        <a
          href="/North America"
          onClick={[Function]}
        >
          <svg
            className="btn"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
            />
          </svg>
        </a>
      </div>
    </li>
    <li
      className="lists"
    >
      <img
        alt="countries"
        src="/Images/Oceania.svg"
      />
      <div
        className="position"
      >
        <h2
          className="title-name"
        >
          Oceania
        </h2>
        <a
          href="/Oceania"
          onClick={[Function]}
        >
          <svg
            className="btn"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
            />
          </svg>
        </a>
      </div>
    </li>
    <li
      className="lists"
    >
      <img
        alt="countries"
        src="/Images/South-America.svg"
      />
      <div
        className="position"
      >
        <h2
          className="title-name"
        >
          South America
        </h2>
        <a
          href="/South America"
          onClick={[Function]}
        >
          <svg
            className="btn"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
            />
          </svg>
        </a>
      </div>
    </li>
  </ul>
</div>
`;
